floss/main.py

```diff
@@ -283,6 +284,54 @@ def set_color_config(color):
         raise RuntimeError("unexpected --color value: " + color)
 
 
+def is_go_bin(sample: str) -> bool:
+    """
+    Check if the binary given is compiled with GO compiler or not
+    it checks the magic header of the pclntab structure -pcHeader-
+    the magic values varies through the version
+    refernce:
+    https://github.com/0xjiayu/go_parser/blob/865359c297257e00165beb1683ef6a679edc2c7f/pclntbl.py#L46
+    """
+    try:
+        pe = pefile.PE(sample)
+    except:
+        logger.debug("NOT valid PE header")
+        return False
+
+    go_magic = [
+        b"\xf0\xff\xff\xff\x00\x00",
+        b"\xfb\xff\xff\xff\x00\x00",
+        b"\xfa\xff\xff\xff\x00\x00",
+        b"\xf1\xff\xff\xff\x00\x00",
+    ]
+
+    for magic in go_magic:
+        for section in pe.sections:
+            section_va = section.VirtualAddress
+            section_size = section.SizeOfRawData
+            # TODO: we can customize the serch to start with .rdata section.
```

Comment:
right, we can probably optimize and skip various sections here

Reply:
I added search for the .data section first. if not available go through all sections